### ISSUE TITLE
Expo bare + viroReact getting this error on ios ->   requireNativeComponent: "VRTText" was not found in the UIManager. 

### ISSUE THREAD

**Issue by byyoungjin** [ViroCommunity/viro#52]
# Requirements: 
Please go through this checklist before opening a new issue
  - [x] Review the [documentation](https://docs.viromedia.com/)
  - [x] Search for existing issues in: [viromedia/viro](https://github.com/viromedia/viro/issues) & [ViroCommunity/viro](https://github.com/ViroCommunity/viro/issues)
  - [x] Use the latest [ViroReact release](https://docs.viromedia.com/docs/releases)

## Environment
Please provide the following information about your environment:
1. Development OS: Mac
2. Device OS & Version: ios 14.7.1
3. Version: 
```
 "dependencies": {
    "@viro-community/react-viro": "^2.21.1",
    "expo": "~42.0.1",
    "expo-splash-screen": "~0.11.2",
    "expo-status-bar": "~1.0.4",
    "expo-updates": "~0.8.1",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "react-native": "~0.63.4",
    "react-native-gesture-handler": "~1.10.2",
    "react-native-reanimated": "~2.2.0",
    "react-native-screens": "~3.4.0",
    "react-native-unimodules": "~0.14.5",
    "react-native-web": "~0.13.12"
  },
  ``` 

4. Device(s): iphone 8 

## Description
I just trying to initialize my app with Expo bare work flow + @viro-community/react-viro 

But keep getting these error 
<img src="https://user-images.githubusercontent.com/44938459/133534359-3f7c98fe-cfef-4cef-b22f-6b4c89c51b7b.jpeg" width="500">

## Reproducible Demo
you can reproduce this by following this 

1. npm install --global expo-cli ( my expo-cli version is 4.11.0)
2. expo init <project name> -> select bare work flow 
3. npm install 
4. npm install --save @viro-community/react-viro 
5. go ios/PodFile and add these lines as instructed on https://github.com/ViroCommunity/viro/blob/main/readmes/INSTALL_IOS.md
pod 'ViroReact', :path => '../node_modules/@viro-community/react-viro/ios/'
pod 'ViroKit_static_lib', :path => '../node_modules/@viro-community/react-viro/ios/dist/ViroRenderer/static_lib'
 
6. npx pod-install 
7. put some viro react code on App.js 
8. npx react-native run ios ( just to check the linking work well) 
-> have error 
Invariant Violation: requireNativeComponent: "VRTText" was not found in the UIManager.
 
9. run xcode, and excute with real device 
->same error with 8

**Comment by creimers** [comment]
I'm experiencing the exact same error as @byyoungjin, but with an app that was bootstrapped with `react-native-cli`. It would appear that something's not quite right with the iOS installation. I'm on react native 0.65.1.

**Comment by Hunnainwali** [comment]
Facing same error on bare work flow. Android is working alright but iOS is showing the UIManager error.

**Comment by danieldulcettiBN** [comment]
I have this problem too, but in my case the UIManager error reports 'VRTARScene' as not found.
Tried in an existing project and in a new one, both created from react-native-cli.

The problem is only for iOS, Android is working well; I have followed all setup points.

React native: 0.65.1
Viro: 2.21.1 (tried also with 2.21.0)
XCode: 12.5 (tried also on XCode 13)

**Comment by Hunnainwali** [comment]
I've got it working. 

1. uninstall viro using $npm uninstall @viro-community/react-viro    
2. install again using  $npm install @viro-community/react-viro@2.20.2 --save (I'm using react-native version 0.64.2)
3. run $npx react-native link
4. run $cd ios
5. run $pod reintegrate
6. run $pod clean
7. run $pod setup
8. run $pod install

For simulator you'll get error related to ARWorldTrackingConfiguration which means project is configured correctly with Viro and you can test on real device. 

Please let me know if it works or not.

**Comment by creimers** [comment]
<img width="520" alt="Bildschirmfoto 2021-09-23 um 12 19 30" src="https://user-images.githubusercontent.com/6090492/134491072-6fafeda2-a9f8-4fd6-86a9-fa81437ce5da.png">

**Comment by mariansam** [comment]
Thanks @Hunnainwali, I hope this will help us. But now we need to find the reason things got broken in [v2.21.0](https://github.com/ViroCommunity/viro/releases/tag/v2.21.0).

Cc @doranteseduardo

**Comment by doranteseduardo** [comment]
Question, is you use the stardard lib instead the static_lib, does it work well or it seems to be broken too?

**Comment by byyoungjin** [comment]
> I've got it working.
> 
> 1. uninstall viro using $npm uninstall @viro-community/react-viro
> 2. install again using  $npm install @viro-community/react-viro@2.20.2 --save (I'm using react-native version 0.64.2)
> 3. run $npx react-native link
> 4. run $cd ios
> 5. run $pod deintegrate
> 6. run $pod clean
> 7. run $pod setup
> 8. run $pod install
> 
> For simulator you'll get error related to ARWorldTrackingConfiguration which means project is configured correctly with Viro and you can test on real device.
> 
> Please let me know if it works or not.

Yes it is working! Thanks. And also work on version 2.21.0.  This problem only happens on 2.21.1 as https://github.com/ViroCommunity/viro/issues/53 says

**Comment by byyoungjin** [comment]
> Question, is you use the stardard lib instead the static_lib, does it work well or it seems to be broken too?

I think I'm using static_lib following this https://github.com/ViroCommunity/viro/blob/main/readmes/INSTALL_IOS.md 
I haven't used standart lib. I Think.

**Comment by Cchumi** [comment]
> I've been unable to build the static_lib, however the standard lib works well for me.
> 
> Can anyone confirm that changing the line:
> 
>  `pod 'ViroKit_static_lib', :path => '../node_modules/@viro-community/react-viro/ios/dist/ViroRenderer/static_lib'`
> 
> to
> 
> `pod 'ViroKit', :path => '../node_modules/@viro-community/react-viro/ios/dist/ViroRenderer/'`
> 
> works?

I don't know this I try it tomorrow and post back the result. 
Thanks 👌

**Comment by HugoGaechter** [comment]
Hello, I have the same issue as you, and changing the line: 
```
pod 'ViroKit_static_lib', :path => '../node_modules/@viro-community/react-viro/ios/dist/ViroRenderer/static_lib'
```
to
```
pod 'ViroKit', :path => '../node_modules/@viro-community/react-viro/ios/dist/ViroRenderer/'
```

didn't work for me either.

I need virovideo therefore I need the 2.21.1 version

@Piquart @Cchumi 
Could you explain to me how to compile the libviroreact library on my own, or send me the dist folder ?

Thank you very much

**Comment by doranteseduardo** [comment]
Closed in favour of #56
